selftests/unit/test_datadir.py

```diff
@@ -24,8 +24,10 @@ def _get_temporary_dirs_mapping_and_config():
         a the path to a configuration file contain those same settings
         """
         base_dir = tempfile.mkdtemp(prefix='avocado_' + __name__)
+        test_dir = os.path.join(base_dir, 'tests')
+        os.mkdir(test_dir)
```

Comment:
Why did you have to change this test? Don't you have to change some where you call the function being changed?

Reply:
The logic is that the system's (and the configured) test directory now has to exist, that is only a user's test directory will be created.  Previously, `avocado.core.data_dir.get_data_dir()` would create it.  For convention only, I'm creating it inside the `base_dir`, but TBH, it could be yet another:

```
test_dir = tempfile.mkdtemp(prefix='avocado_' + __name__)
```

Because the point of this code is to setup configuration file and the directories following it (be them whenever they are).